misk-aws/src/main/kotlin/misk/jobqueue/sqs/AwsSqsJobHandlerModule.kt

```diff
@@ -0,0 +1,63 @@
+package misk.jobqueue.sqs
+
+import com.google.common.util.concurrent.AbstractIdleService
+import com.google.common.util.concurrent.Service
+import com.google.inject.Key
+import misk.DependentService
+import misk.inject.KAbstractModule
+import misk.jobqueue.JobHandler
+import misk.jobqueue.QueueName
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.reflect.KClass
+
+/**
+ * Install this module to register a handler for an SQS queue.
+ */
+class AwsSqsJobHandlerModule<T : JobHandler> private constructor(
+  private val queueName: QueueName,
+  private val handler: KClass<T>
+) : KAbstractModule() {
+  override fun configure() {
+    newMapBinder<QueueName, JobHandler>().addBinding(queueName).to(handler.java)
+    multibind<Service>().to<AwsSqsJobHandlerSubscriptionService>()
+  }
+
+  companion object {
+    inline fun <reified T : JobHandler> create(queueName: QueueName):
+        AwsSqsJobHandlerModule<T> = create(queueName, T::class)
+
+    @JvmStatic
+    fun <T : JobHandler> create(
+      queueName: QueueName,
+      handlerClass: Class<T>
+    ): AwsSqsJobHandlerModule<T> {
+      return create(queueName, handlerClass.kotlin)
```

Comment:
nit: weird to convert to KT to convert back to java. I'd just keep it java, since guice needs that anyways for binding

Reply:
It seems like we usually convert to KClass throughout misk.